src/Compilers/CSharp/Portable/Parser/LanguageParser_InterpolatedString.cs

```diff
@@ -39,98 +40,234 @@ private ExpressionSyntax ParseInterpolatedStringToken()
             var originalToken = this.EatToken();
 
             var originalText = originalToken.ValueText; // this is actually the source text
+            var originalTextSpan = originalText.AsSpan();
             Debug.Assert(originalText[0] == '$' || originalText[0] == '@');
 
             // compute the positions of the interpolations in the original string literal, if there was an error or not,
-            // and where the close quote can be found.
+            // and where the open and close quotes can be found.
             var interpolations = ArrayBuilder<Lexer.Interpolation>.GetInstance();
 
             rescanInterpolation(out var kind, out var error, out var openQuoteRange, interpolations, out var closeQuoteRange);
 
-            var result = SyntaxFactory.InterpolatedStringExpression(
-                getOpenQuote(openQuoteRange), getContent(interpolations), getCloseQuote(closeQuoteRange));
+            // Only bother trying to do dedentation if we have a multiline literal without errors.  There's no point
+            // trying in the presence of errors as we may not even be able to determine what the dedentation should be.
+            var isMultiLineWithoutErrors = kind == Lexer.InterpolatedStringKind.MultiLineRaw && error == null;
```

Comment:
>isMultiLineWithoutErrors [](http://example.com/codeflow?start=16&length=24)

nit: if this local stays (see other comments) consider naming `needsDedentation`

Reply:
done!